src/smtp_socket.erl

```diff
@@ -120,13 +120,24 @@ accept(Socket, Timeout) when is_port(Socket) ->
 		Error ->
 			Error
 	end;
+
+-ifdef(deprecated_ssl_accept).
 accept(Socket, Timeout) ->
 	case ssl:transport_accept(Socket, Timeout) of
 		{ok, NewSocket} ->
 			ssl:ssl_accept(NewSocket),
 			{ok, NewSocket};
 		Error -> Error
 	end.
+-else.
+accept(Socket, Timeout) ->
+	case ssl:transport_accept(Socket, Timeout) of
+		{ok, NewSocket} ->
+			ssl:handshake(NewSocket),
+			{ok, NewSocket};
+		Error -> Error
+	end.
+-endif.
```

Comment:
another option would be to create a new function and use it in both places:

```
-ifdef(deprecated_ssl_accept).
ssl_handshake(Sock) ->
    ssl:handshake(Sock).
-else.
ssl_handshake(Sock) ->
    ssl:ssl_accept(NewSocket).
-endif.
```
This way you will have less code duplication.

Reply:
Yes I agree with @seriyps here, less code duplication = better